rrmngmnt/storage.py

```diff
@@ -0,0 +1,183 @@
+            rc = self.host.fs.rmdir(mount_point)
+            if not rc:
+                self.logger.error("failed to remove directory %s", mount_point)
+                return False
+        return True
+
+    def create_file_on_host(self, file_name, path):
+        """
+        Creates a file on host
+
+        __author__ = "ratamir"
+        :param file_name: The file to create
+        :type file_name: str
+        :param path: The path that the file will create under
+        :type path: str
+        :returns: True if succeeded in creating file requested,
+        False otherwise
+        :rtype: bool
+        """
+        command = CREATE_FILE_CMD % os.path.join(path, file_name)
+        return self.host.run_command(shlex.split(command))[0]
+
+
+class LVMService(Service):
+    """
+    Storage management class to maintain LVM services
+    """
+
+    def lv_change(self, vg_name, lv_name, activate=True):
+        """
+        Set the LV attribute 'active' (active or inactive)
+
+        __author__ = "ratamir, glazarov"
+        :param vg_name: The name of the Volume group under which LV is
+        contained
+        :type vg_name: str
+        :param lv_name: The name of the logical volume which needs to be
+        activated or deactivated
+        :type lv_name: str
+        :param activate: True in case the lv change command should activate
+        the ls, False in case of deactivate it
+        :type activate: bool
+        :returns: True if succeeded, False otherwise
+        :rtype: bool
+        """
+        active = 'y' if activate else 'n'
+
+        logger.info(
+            "Setting the logical volume 'active=%s' attribute", activate
+        )
+        rc, output, error = self.host.run_command(
+            shlex.split(LV_CHANGE_CMD % active, vg_name, lv_name)
+        )
+        if rc:
+            logger.error(
+                "Error while setting the logical volume 'active=%s' "
+                "attribute. Output is '%s', error is '%s'",
+                activate, output, error
+            )
+            return False
+        return True
+
+    def run_pvscan_command(self):
+        """
+        Execute 'pvscan --cache' in order to get the latest list
+        of volumes. In case no data is returned (equivalent to no
+        changes), run 'pvscan' by itself.
+        This combination appears to correctly retrieve the latest
+        volume listing
+
+        __author__ = "ratamir, glazarov"
+        :returns: True if succeeded, False otherwise
+        :rtype: bool
+        """
+        logger.info("Executing '%s' command", PVSCAN_CACHE_CMD)
+        rc, output, error = self.host.run_command(
+            shlex.split(PVSCAN_CACHE_CMD)
+        )
+        if rc:
+            logger.error(
+                "Error while executing the '%s' command, output is '%s', "
+                "error is '%s'", PVSCAN_CACHE_CMD, output, error
+            )
+            return False
+
+        if output == '':
```

Comment:
This is not looking good, maybe try to find a better why to get all volumes.
when using run_command there is no need for logger.info (you added it to the function)